### PR TITLE
Graph refinement

### PR DESCRIPTION
**Pull request by RyKoe**
Added functionality for density plots. 
- One plot for top 5 of category
- One plot for selected list
- Added bool in function to switch between
- If no selection is made will default to top 5 category even if bool is set to selected list plot
- Added ability to change year range of scatter and count plots (if no year_in or year_out defaulted to entire range)

### CODE REVIEW

File: src/app/app_graphing.py [hunk 1]
```diff
@@ -361,6 +361,66 @@ def graph_3D(data, col="category", list_=[None], game=None):
...
+
+    data: a pandas df generated from app_wrangling.call_boardgame_data()
+    col: string
+    list_: list
+    year_in: int
+    year_out: int
+
+    return: altair plot
+    """
+    if (bool_ is True) or (not bool(list_)):
```
Comment (WraySmith) [inline src/app/app_graphing.py:383]: `if (bool_ is True) or (not bool(list_)` should be be `if bool_ or ((not bool(list_))`

File: src/app/app_graphing.py [hunk 2]
```diff
@@ -361,6 +361,66 @@ def graph_3D(data, col="category", list_=[None], game=None):
...
+        plot_data = app_wr.call_boardgame_density(data, col, year_in, year_out)
+    else:
+        plot_data = app_wr.call_boardgame_radio(data, col, list_, year_in, year_out)
+
+    rank_plot = (
+        alt.Chart(plot_data, height=80)
+        .transform_density(
+            "average_rating", as_=["average_rating", "density"], groupby=["group"]
+        )
```
Comment (WraySmith) [inline src/app/app_graphing.py:392]: the experimental Altair plot transformer may need to be used for this plot if it's slow in the app

File: src/app/app_graphing.py [hunk 3]
```diff
@@ -361,6 +361,66 @@ def graph_3D(data, col="category", list_=[None], game=None):
...
+        .encode(
+            alt.X("average_rating:Q", bin="binned", title="Average Rating"),
+            alt.Y(
+                "density:Q", axis=None, title=None, scale=alt.Scale(domain=[0, 0.75])
+            ),
+            alt.Color("group:N", title=None, scale=alt.Scale(scheme="set3")),
+        )
+    )
+
+    avg_line = (
```
Comment (WraySmith) [inline src/app/app_graphing.py:405]: the avg calc should potentially be brought out of the plot (unless we're going to the experimental Altair transformer)

File: src/app/app_wrangling.py [hunk 4]
```diff
@@ -190,7 +192,7 @@ def form_group(data, col, list_):
 
     # replaces groups containing all items with 'All Selected'
     if len(list_) > 1:
-        data["group"] = data["group"].apply(lambda x: helper_form_group(x, list_))
+        data.group = data["group"].apply(lambda x: helper_form_group(x, list_))
```
Comment (WraySmith) [inline src/app/app_wrangling.py:196]: why the change?

File: src/app/app_wrangling.py [hunk 5]
```diff
@@ -311,3 +304,60 @@ def remove_columns(data):
...
+    boardgame_data = data.copy(deep=True)
+    boardgame_list = call_boardgame_top(data, col, year_in, year_out)[col].to_list()
+
+    boardgame_data = boardgame_data[
+        call_bool_series_or(boardgame_data, col, boardgame_list)
+    ]
+    boardgame_data = form_group(boardgame_data, col, boardgame_list)
+    boardgame_data = boardgame_data.explode("group")
+
+    return boardgame_data
```
Comment (WraySmith) [inline src/app/app_wrangling.py:335]: will `group` always exist when this is called?

File: src/app/app_graphing.py [hunk 6]
```diff
@@ -361,6 +361,66 @@ def graph_3D(data, col="category", list_=[None], game=None):
...
+    year_in: int
+    year_out: int
+
+    return: altair plot
+    """
+    if (bool_ is True) or (not bool(list_)):
+        plot_data = app_wr.call_boardgame_density(data, col, year_in, year_out)
+    else:
+        plot_data = app_wr.call_boardgame_radio(data, col, list_, year_in, year_out)
+
```
Comment (WraySmith) [inline src/app/app_graphing.py:387]: what's the difference between the if statement output - seems like a pretty big difference between `call_boardgame_density()` and `call_boardgame_radio()`?

File: src/app/app_wrangling.py [hunk 7]
```diff
@@ -311,3 +304,60 @@ def remove_columns(data):
         keep.append("group")
 
     return boardgame_data[keep]
+
+
+def call_boardgame_density(data, col, year_in, year_out):
+    """
```
Comment (WraySmith) [inline src/app/app_wrangling.py:310]: this function should potentially be renamed something along the lines of `call_boardgame_top_density()` to be more clear

File: src/app/app_graphing.py (hunk 1 above)
Comment (RyKoe) [reply src/app/app_graphing.py:383]: Very true.  Over complicated it

**Comment by WraySmith** [comment]
A couple comments and questions. I also tried running the app and got a couple errors that should be resolved before merge in.

File: src/app/app_wrangling.py (hunk 4 above)
Comment (RyKoe) [reply src/app/app_wrangling.py:196]: Oops that wasn't supposed to have remained.  I was getting warning out of this function on setting values by using slice instead of .loc[]  I tried a couple things to get them to stop, but no luck. Thought I had reverted changes

File: src/app/app_wrangling.py (hunk 5 above)
Comment (RyKoe) [reply src/app/app_wrangling.py:335]: It should.  It only is called when a selection list is provided and the form_group function should always create a group column

File: src/app/app_graphing.py (hunk 3 above)
Comment (RyKoe) [reply src/app/app_graphing.py:405]: I'll look at doing this in the next PR

**Review by WraySmith — APPROVED** [review]
